File: managementSystem/react-management/src/pages/Item/CampEquipment.js
import axios from 'axios';
import {
  MaterialReactTable,
  useMaterialReactTable,
} from 'material-react-table';
import React, { useEffect, useState } from 'react';

const CampEquipment = () => {
  const [data, setData] = useState([]);

  useEffect(() => {
    axios
      .get('/manage/item?categoryCode=2')
      .then((res) => {
        setData(res.data.itemList);
      })
      .catch((error) => {
        console.log('error');
      });
  }, []);

  const columns = [
    {
      accessorKey: 'itemNo',
      header: '상품 번호',
    },
    {
      accessorKey: 'categoryName',
      header: '카테고리 명',
    },
    {
      accessorKey: 'equipmentCategoryName',
      header: '캠핌용품 카테고리',
    },
    {
      accessorKey: 'equipmentName',
      header: '캠핌용품 명',
    },
    {
      accessorKey: 'equipmentRentPrice',
      header: '대여 비용',
    },
    {
      accessorKey: 'equipmentRentCount',
      header: '대여 재고',
    },
    {
      accessorKey: 'equipmentSellPrice',
      header: '판매 가격',
    },
    {
      accessorKey: 'equipmentSellCount',
      header: '판매 재고',
    },
  ];

  const table = useMaterialReactTable({
    columns,
    data,
    isMultiSortEvent: () => true,
  });

  return <MaterialReactTable table={table} />;
};

export default CampEquipment;

File: wheelingcamp/src/main/java/kr/co/wheelingcamp/item/model/dto/CampEquipment.java
package kr.co.wheelingcamp.item.model.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 * 캠핑용품 관리 DTO
 */
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class CampEquipment extends Item {

	private int equipmentCategoryCode; // 캠핑용품 카테고리 번호
	private String equipmentCategoryName; // 캠핑용품 카테고리 이름

	private String equipmentName; // 캠핑용품 이름

	// 캠핑용품 대여
	private String equipmentRentPrice; // 캠핑용품 대여 비용
	private int equipmentRentCount; // 캠핑용품 대여 재고

	// 캠핑용품 판매
	private int equipmentSellPrice; // 캠핑용품 판매 가격
	private int equipmentSellCount; // 캠핑용품 판매 재고
}
